mlx/backend/metal/kernels/CMakeLists.txt

```diff
@@ -93,6 +107,7 @@ if(NOT MLX_METAL_JIT)
   build_kernel(steel/gemm/kernels/steel_gemm_fused ${STEEL_HEADERS})
   build_kernel(steel/gemm/kernels/steel_gemm_masked ${STEEL_HEADERS})
   build_kernel(steel/gemm/kernels/steel_gemm_splitk ${STEEL_HEADERS})
+  build_kernel(steel/attn/kernels/steel_attention ${STEEL_ATTN_HEADERS})
```

Comment:
I think we should optionally put this in the JIT (can be a follow on). Right now the behavior is just to build if the jit is not enabled. If the JIT is enabled then the kernel won't be there. You can move it out of this condition to get it to pass the tests for now.